packages/instantsearch-ui-components/src/types/Renderer.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable @typescript-eslint/no-namespace */
+/* eslint-disable @typescript-eslint/no-empty-interface */
+
+// Ensure that the JSX namespace is available in this file and its dependents.
+declare global {
+  namespace JSX {
+    interface Element {}
+    interface IntrinsicElements {}
+  }
+}
+
+// For safety when there's no @types/react or preact, we don't directly use
+// JSX.IntrinsicElements
+type IntrinsicElements = keyof JSX.IntrinsicElements extends never
```

Comment:
in the future we probably want to get rid of these types and make everything a bit more generic / less focused on react etc. For now I think it's generic enough as Vue isn't in typescript anyway